core/src/test/java/com/linecorp/armeria/client/circuitbreaker/CircuitBreakerClientRuleTest.java

```diff
@@ -107,14 +122,87 @@ void openCircuitWithTrailer() {
                          .build();
 
         assertThat(client.get("/trailers").aggregate().join().trailers()).containsExactly(
-                Maps.immutableEntry(HttpHeaderNames.of("grpc-status"), "3"));
+                Maps.immutableEntry(HttpHeaderNames.of("x-checksum"), "42"));
         await().untilAsserted(() -> {
             assertThatThrownBy(() -> client.get("/trailers").aggregate().join())
                     .isInstanceOf(CompletionException.class)
                     .hasCauseInstanceOf(FailFastException.class);
         });
     }
 
+    @Test
+    void openCircuitWithGrpc() {
+        final CircuitBreakerRule rule =
+                CircuitBreakerRule.builder()
+                                  .onGrpcTrailers((ctx, trailers) -> trailers.containsInt("grpc-status", 3))
+                                  .thenFailure();
+
+        final BlockingWebClient client =
+                WebClient.builder(server.httpUri())
+                         .decorator(CircuitBreakerClient.newDecorator(CircuitBreaker.ofDefaultName(), rule))
```

Comment:
How about configuring the `CircuitBreaker` to fail fast?

Reply:
Could you also add a test that uses `onGrpcTrailers` in RetryingClient, please?

Reply:
That's a good idea. I added a new factory method `superSensitiveCircuitBreaker()` and used it for all tests in `CircuitBreakerClientRuleTest`.

I also modified `GrpcClientRetryTest` to use `onGrpcTrailers()`, which resolves your second comment.